jscomp/main/jsoo_refmt_main.ml

```diff
@@ -0,0 +1,251 @@
+
+let export (field : string) v =
+  Js.Unsafe.set (Js.Unsafe.global) field v
+;;
+
+(* To add a directory to the load path *)
+
+let dir_directory d =
+  Config.load_path := d :: !Config.load_path
+
+
+let () =
+  dir_directory "/static/cmis"
+
+module Converter = Refmt_api.Migrate_parsetree.Convert(Refmt_api.Migrate_parsetree.OCaml_404)(Refmt_api.Migrate_parsetree.OCaml_406)
+
+let reason_parse lexbuf = 
+  Refmt_api.Reason_toolchain.RE.implementation lexbuf |> Converter.copy_structure;;
+
+let make_compiler name impl prefix =
+  export name
+    (Js.Unsafe.(obj
+                  [|"compile",
+                    inject @@
+                    Js.wrap_meth_callback
+                      (fun _ code ->
+                         (compile impl ~use_super_errors:false (Js.to_string code)));
+                    "shake_compile",
+                    inject @@
+                    Js.wrap_meth_callback
+                      (fun _ code ->
+                         (shake_compile impl ~use_super_errors:false (Js.to_string code) prefix));
+                    "compile_super_errors",
+                    inject @@
+                    Js.wrap_meth_callback
+                      (fun _ code ->
+                         (compile impl ~use_super_errors:true (Js.to_string code)));
+                    "compile_super_errors_ppx_v2",
+                    inject @@
+                    Js.wrap_meth_callback
+                      (fun _ code ->
+                         (compile impl ~use_super_errors:true ~react_ppx_version:V2 (Js.to_string code)));
+                    "compile_super_errors_ppx_v3",
+                    inject @@
+                    Js.wrap_meth_callback
+                      (fun _ code ->
+                         (compile impl ~use_super_errors:true ~react_ppx_version:V3 (Js.to_string code)));
+                    "shake_compile_super_errors",
+                    inject @@
+                    Js.wrap_meth_callback
+                      (fun _ code -> (shake_compile impl ~use_super_errors:true (Js.to_string code) prefix));
+                    "version", Js.Unsafe.inject (Js.string (match name with | "reason" -> Refmt_api.version | _ -> Bs_version.version));
+                    "load_module",
+                    inject @@
+                    Js.wrap_meth_callback
+                      (fun _ cmi_path cmi_content cmj_name cmj_content ->
+                        let cmj_bytestring = Js.to_bytestring cmj_content in
+                        (* HACK: force string tag to ASCII (9) to avoid
+                         * UTF-8 encoding *)
+                        Js.Unsafe.set cmj_bytestring "t" 9;
+                        load_module cmi_path cmi_content (Js.to_string cmj_name) cmj_bytestring);
+                  |]))
+
+let () = make_compiler "ocaml" Parse.implementation "[@@@bs.config{no_export}]\n#1 \"repl.ml\"\n"
+let () = make_compiler "reason" reason_parse "[@bs.config {no_export: no_export}];\n#1 \"repl.re\";\n"
+
+let _ = 1
+
```

Comment:
why `let _ = 1` here?

Reply:
that was a left over from testing the compiling process... removed